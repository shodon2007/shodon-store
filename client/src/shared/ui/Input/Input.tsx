import { FC, InputHTMLAttributes } from 'react';
import classNames from 'src/shared/lib/classNames/classNames';
import cls from './Input.module.scss';
interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
	className?: string;
}
const Input: FC<InputProps> = ({ className, ...props }) => {
	return (
		<input
			data-testid='input'
			{...props}
			className={classNames(cls.Input, {}, [className])}
		/>
	);
};
export default Input;
