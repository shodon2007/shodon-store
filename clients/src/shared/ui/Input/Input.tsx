import { FC, InputHTMLAttributes } from "react";
import classNames from "/shared/lib/classNames";
import cls from "./Input.module.scss";
interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
    className?: string;
}
const Input: FC<InputProps> = ({ className, ...props }) => {
    return (
        <input {...props} className={classNames(cls.Input, {}, [className])} />
    );
};
export default Input;