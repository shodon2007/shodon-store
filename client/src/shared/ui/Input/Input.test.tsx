import { render, screen } from "@testing-library/react"
import Input from "./Input"

describe('testing Input.tsx', () => {
    test('testing input className', () => {
        render(<Input className='hello world' />);
        const input = screen.getByTestId('input');
        expect(input).toBeInTheDocument();
    })
})