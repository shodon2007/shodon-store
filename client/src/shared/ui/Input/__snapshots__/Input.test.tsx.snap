// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Input.tsx testing input className 1`] = `
<input
  class="Input hello world"
  data-testid="input"
/>
`;
